src/test/java/org/codehaus/mojo/license/LicenseMojoUtilsTest.java

```diff
@@ -0,0 +1,162 @@
+package org.codehaus.mojo.license;
+
+import static org.junit.Assert.*;
+
+import java.io.File;
+
+import org.junit.Test;
+
+public class LicenseMojoUtilsTest
+{
+    private String resolvedUrl;
+    private File deprecatedFile;
+    private String url;
+    private File basedir = new File( "" );
+    private MockLogger log = new MockLogger();
+
+    @Test
+    public void testIsValidNull()
+    {
+        assertFalse( LicenseMojoUtils.isValid( null ) );
+    }
+
+    @Test
+    public void testIsValidEmpty()
+    {
+        // TODO is this correct?
+        assertTrue( LicenseMojoUtils.isValid( "" ) );
+    }
+
+    @Test
+    public void testIsValidBlank()
+    {
+        // TODO is this correct?
+        assertTrue( LicenseMojoUtils.isValid( "   " ) );
+    }
+
+    @Test
+    public void testIsValidNonexistingClasspathResource()
+    {
+        assertTrue( LicenseMojoUtils.isValid( "classpath:noSuchResource" ) );
+    }
+
+    @Test
+    public void testIsValidClasspathResource()
+    {
+        assertTrue( LicenseMojoUtils.isValid( "classpath:log4j.properties" ) );
+    }
+
+    @Test
+    public void testIsValidHttpResource()
+    {
+        assertTrue( LicenseMojoUtils.isValid( "http://foo/bar/baz" ) );
+    }
+
+    @Test
+    public void testPrepareThirdPartyOverrideUrlNull()
+    {
+        LicenseMojoUtils.prepareThirdPartyOverrideUrl( resolvedUrl, deprecatedFile, url, basedir, log );
+    }
+    
+    @Test
+    public void testPrepareThirdPartyOverrideUrlBothOverrides()
+    {
+        deprecatedFile = new File( "src/test/resources/overrides.properties" );
+        url = "classpath:overrides.properties";
+        try
+        {
+            LicenseMojoUtils.prepareThirdPartyOverrideUrl( resolvedUrl, deprecatedFile, url, basedir, log );
+        }
```

Comment:
If the IllegalArgumentException is expected, the test should fail in case the exception is not thrown. So you probably want to add something like `Assert.fail(IllegalArgumentException.class.getSimpleName() + " expected")` on line 69

Reply:
Done.